### ISSUE TITLE
Investigate into Text methods

### ISSUE THREAD

**Issue by gossi** [phootwork/lang#2]
Run some research, which methods other frameworks use and search for consistency among them.

**Comment by cristianoc72** [comment]
Getting a look around, we can find 3 kinds of approach:
1.  frameworks inspired on Java or Python string objects (like this library), which implement the original methods as much as possible
2.  "macedonia" libraries, which wrap all php string functions in an OOP way, plus various methods the owner thinks useful (i.e. https://github.com/danielstjules/Stringy)
3.  libraries splitted from full-stack frameworks, which have very specific purpose (i.e. Laravel helpers, thought to manipulate html)

Imho, this library is already good and I'ld only like to add the following:
-   some issers (isAlnum, isNumeric, .....) taken from Python string object https://docs.python.org/3/library/stdtypes.html.
-  some methods to convert to snake case or camel case (I'm thinking about Propel NamingTool). Almost all libraries have something similiar.

Other methods, included in "macedonia" libraries, imho should be implemented in a formatter class. This is the Python way, for example: the string object has a `format` method which internally uses a `Formatter` object.

**Comment by gossi** [comment]
Good one. I also have `NameTools` with further utilities I use at various places, they are:

```php
class NameTools {
  public static toStudlyCase();
  public static toCamelCase();
  public static toKebapCase();
  public static toSnakeCase();
  public static pluzalize();
  public static singularize();
}
```

Dunno if the latter two would make a good addition to the `Text` class, though the first four do (pluralize/singularize do use the propel Pluralizer under the hood =). 

Would you mind a PR with good additions?

**Comment by cristianoc72** [comment]
Sure!
Since Propel Pluralizer is very good andit's not strictly related to Propel, I think we should split it in a standalone library.
Let's go!

**Comment by gossi** [comment]
While you are at it, could you also have a look at the current method naming and come up with a naming scheme? E.g. here are `to*Case()` methods, while currently there are `lower()`,  `upper()`, `capitalize()`. Might be good that all "transforms" start with a `to` prefix (which also makes it handy for auto-completion) and all "checks" to begin with `is` for the same reason? That would be my background thinking of this. What do you think?

Anyway, we would have a class, that has `lower()`, `capitalize()`, `dasherize()`, `studlyize()` (is that word even a thing?) and so forth, might be hard to grasp what all the utilities be, just being in the IDE given an instance of the `Text` class.

Given the current methods, they become aliases of the new ones, with a stub call inside and a `@deprecated` on top?

**Comment by cristianoc72** [comment]
Completely agree! :+1:

**Comment by gossi** [comment]
Done, thanks to the great PR by @cristianoc72 
I also just added multibyte support. I keep this open, since I need tests for this.

**Comment by gossi** [comment]
Well, I dropped it ... still putting in more methods, `Text` is becoming a swiss army knife I guess but it's a good thing.

**Comment by gossi** [comment]
Released `v0.9` is out. Thanks a lot @cristianoc72

**Comment by cristianoc72** [comment]
Thank you! :satisfied:
